src/Microsoft.AspNetCore.Mvc.Razor/Internal/ViewComponentTagHelperDescriptorFactory.cs

```diff
@@ -56,7 +57,6 @@ public IEnumerable<TagHelperDescriptor> CreateDescriptors(string assemblyName)
                     StringComparison.Ordinal));
 
             var tagHelperDescriptors = viewComponentDescriptors
-                .Where(descriptor => !descriptor.Parameters.Any(parameter => parameter.ParameterType.GetTypeInfo().IsGenericType))
```

Comment:
Is the method found simply because its name is InvokeAsync? If so, shouldn't the fix be to check `!memberInfo.IsGenericMethod` or similar?